processor/k8sattributesprocessor/processor_test.go

```diff
@@ -319,6 +346,34 @@ func TestIPDetectionFromContext(t *testing.T) {
 		require.Greater(t, r.Attributes().Len(), 0)
 		assertResourceHasStringAttribute(t, r, "k8s.pod.ip", "1.1.1.1")
 	})
+
+}
+
+func TestIPDetectionFromUDPContext(t *testing.T) {
+	m := newMultiTest(t, NewFactory().CreateDefaultConfig(), nil)
+
+	udpCtx := client.NewContext(context.Background(), client.Info{
```

Comment:
I think that iterating the test over TCPAddr, UDPAddr and IPAddr could be somewhat helpful to make sure only the IP address part is extracted in each case

Reply:
agreed - i simplified this to iterate over a slice of net.Addr